flowise-embed/src/components/Bot.tsx

```diff
@@ -1023,6 +1057,62 @@ export const Bot = (botProps: BotProps & { class?: string }) => {
     }
   };
 
+  const isNCM = async (ncm: string): Promise<boolean> => {
+    const ncmFormatted = ncm.replace(/\./g, '').replace(/\D/g, '');
+    if (ncmFormatted.length !== 8) {
+      setMessages((prevMessages) => {
+        const newMessage = { message: ncmLengthErrorMessage(ncm), type: 'apiMessage' } as MessageType;
+        const updated = [...prevMessages, newMessage];
+        addChatMessage(updated);
+        return [...updated, { message: '', type: 'apiMessage' }];
+      });
+      return false;
+    }
+    const result = await handleNcmExistence(ncmFormatted);
+    if (result?.isNCM === false) {
+      setMessages((prevMessages) => {
+        const newMessage = { message: ncmExistenceErrorMessage(ncm), type: 'apiMessage' } as MessageType;
+        const updated = [...prevMessages, newMessage];
+        addChatMessage(updated);
+        return [...updated, { message: '', type: 'apiMessage' }];
+      });
+      return false;
+    }
+    if (JSON.stringify(result) === '{}') {
+      setMessages((prevMessages) => {
+        const newMessage = { message: ncmValidationErrorMessage(ncm), type: 'apiMessage' } as MessageType;
+        const updated = [...prevMessages, newMessage];
+        addChatMessage(updated);
+        return [...updated, { message: '', type: 'apiMessage' }];
+      });
+      return true;
```

Comment:
Não entendi - se for um objeto vazio dá mensagem de erro e retorna `true`? 🤔

Reply:
Essa lógica parece ir um pouco além do que está descrito nesse trecho de código huahua
No caso do código, retornar `true` indica o quê, no processo como um todo?